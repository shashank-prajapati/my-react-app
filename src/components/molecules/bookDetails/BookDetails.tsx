import React from "react";
import Box from "@mui/material/Box";
import Typography from "@mui/material/Typography";
import TimeLeft from "../timeLeft/TimeLeft";
import { ReactComponent as HorizontalMenu } from "../../../media/horizontalMenu.svg";
import BookCardButton from "../../atoms/bookCardButton/BookCardButton";
import { ButtonProps } from "@mui/material";
import Icon from "../../atoms/icon/Icon";

interface Props extends ButtonProps {
  time: string;
  bookTitle: string;
  bookAuthor: string;
  isButtoned: Boolean;
  btnText?: string;
  isHovered: Boolean;
}

function BookDetails(props: Props) {
  const {
    bookTitle,
    bookAuthor,
    time,
    isButtoned,
    btnText,
    startIcon,
    isHovered,
  } = props;
  let backgroundColor = "white";
  if (isHovered) {
    backgroundColor = "footerBgColor.main";
  }
  return (
    <div>
      <Box
        sx={{
          display: "flex",
          flexDirection: "column",
          flexWrap: "wrap",
          width: "99%",
          mx: "auto",
          borderLeft: 1,
          borderRight: 1,
          borderColor: "secondary_100.main",
          bgcolor: isHovered ? "footerBgColor.main" : "white",
          mt: 0,
        }}
      >
        <Box sx={{ px: 4, pt: 6, pb: 2 }}>
          <Typography variant="subtitle2" color="navTextColors.main">
            {bookTitle}
          </Typography>
          <Typography
            variant="body1"
            color="extraTextColor.main"
            sx={{ mt: 4 }}
          >
            {bookAuthor}
          </Typography>
          <TimeLeft sx={{ mt: 4 }} time={time} />
          <Box
            sx={{
              display: "flex",
              mt: 2,
              justifyContent: "flex-end",
            }}
          >
            {!isButtoned && (
              <Icon iconComponent={HorizontalMenu} width="18px" height="4px" />
            )}
          </Box>
        </Box>
        {isButtoned && (
          <BookCardButton
            btnText={btnText}
            startIcon={startIcon}
            isHovered={isHovered}
          />
        )}
      </Box>
    </div>
  );
}

export default BookDetails;
