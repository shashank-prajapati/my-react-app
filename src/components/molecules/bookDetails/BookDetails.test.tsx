import * as React from "react";
import { render, fireEvent } from "@testing-library/react";
import BookDetails from "./BookDetails";
import { Typography } from "@mui/material";

describe("Testing Book Details Molecule", () => {
  it("should match the unButtoned BookDetails snapshot", () => {
    const { container } = render(
      <BookDetails
        time="13"
        bookAuthor={"Jim Collins & Bill Lazier"}
        bookTitle="Beyond Entrepreneurship"
        isButtoned={false}
        isHovered={false}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("should match the buttoned snapshot", () => {
    const { container } = render(
      <BookDetails
        time="13"
        bookAuthor={"Jim Collins & Bill Lazier"}
        bookTitle="Beyond Entrepreneurship"
        isButtoned={true}
        isHovered={false}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
