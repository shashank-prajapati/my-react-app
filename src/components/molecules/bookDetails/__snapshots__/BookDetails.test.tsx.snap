// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Book Details Molecule should match the buttoned snapshot 1`] = `
<div>
  <div>
    <div
      class="MuiBox-root css-9r861l"
    >
      <div
        class="MuiBox-root css-3jmp7d"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 css-i5c7wg-MuiTypography-root"
        >
          Beyond Entrepreneurship
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-hi830k-MuiTypography-root"
        >
          Jim Collins & Bill Lazier
        </p>
        <div>
          <div
            class="MuiBox-root css-5as2nd"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2v1qra-MuiSvgIcon-root"
              focusable="false"
            >
              clockIcon.svg
            </svg>
            <span
              class="MuiTypography-root MuiTypography-caption3 css-op2c4e-MuiTypography-root"
            >
              13
              -minute read
            </span>
          </div>
        </div>
        <div
          class="MuiBox-root css-eb3voy"
        />
      </div>
      <div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-jbktht-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing Book Details Molecule should match the unButtoned BookDetails snapshot 1`] = `
<div>
  <div>
    <div
      class="MuiBox-root css-9r861l"
    >
      <div
        class="MuiBox-root css-3jmp7d"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 css-i5c7wg-MuiTypography-root"
        >
          Beyond Entrepreneurship
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-hi830k-MuiTypography-root"
        >
          Jim Collins & Bill Lazier
        </p>
        <div>
          <div
            class="MuiBox-root css-5as2nd"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2v1qra-MuiSvgIcon-root"
              focusable="false"
            >
              clockIcon.svg
            </svg>
            <span
              class="MuiTypography-root MuiTypography-caption3 css-op2c4e-MuiTypography-root"
            >
              13
              -minute read
            </span>
          </div>
        </div>
        <div
          class="MuiBox-root css-eb3voy"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-r43foo-MuiSvgIcon-root"
            focusable="false"
          >
            horizontalMenu.svg
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
